homeassistant/components/tesla/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Tesla Config Flow."""
+from collections import OrderedDict, defaultdict
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+DOMAIN = "tesla"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Tesla instances."""
+    return set(entry.title for entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class TeslaFlowHandler(config_entries.ConfigFlow):
+    """Handle a Tesla config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.login = None
+        self.config = OrderedDict()
+        self.data_schema = OrderedDict(
+            [
+                (vol.Required(CONF_USERNAME), str),
+                (vol.Required(CONF_PASSWORD), str),
+                (
+                    vol.Optional(CONF_SCAN_INTERVAL, default=300),
+                    vol.All(cv.positive_int, vol.Clamp(min=300)),
+                ),
+            ]
+        )
+
+    async def _show_form(
+        self, step="user", placeholders=None, errors=None, data_schema=None
+    ) -> None:
+        """Show the form to the user."""
+        _LOGGER.debug("show_form %s %s %s %s", step, placeholders, errors, data_schema)
```

Comment:
Please remove this debug log before merge.

Reply:
Ok. Where's the documentation on when debug logs can be included?